rpc/handlers_test.go

```diff
@@ -889,3 +890,69 @@ func TestStateUpdate(t *testing.T) {
 		}
 	})
 }
+
+func TestSyncing(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	mockReader := mocks.NewMockReader(mockCtrl)
+	handler := rpc.New(mockReader, utils.MAINNET)
+	defaultState := false
+
+	t.Run("undefined starting block", func(t *testing.T) {
+		mockReader.EXPECT().StartingBlockNumber().Return(nil)
+		syncing, err := handler.Syncing()
+		assert.Nil(t, err)
+		assert.Equal(t, &rpc.SyncState{False: &defaultState}, syncing)
+	})
+
+	startingBlock := uint64(0)
+	t.Run("empty blockchain", func(t *testing.T) {
+		mockReader.EXPECT().StartingBlockNumber().Return(&startingBlock)
+		mockReader.EXPECT().BlockHeaderByNumber(startingBlock).Return(nil, errors.New("empty blockchain"))
+
+		syncing, err := handler.Syncing()
+		assert.Nil(t, err)
+		assert.Equal(t, &rpc.SyncState{False: &defaultState}, syncing)
+	})
+	t.Run("undefined highest block", func(t *testing.T) {
+		mockReader.EXPECT().StartingBlockNumber().Return(&startingBlock)
+		mockReader.EXPECT().BlockHeaderByNumber(startingBlock).Return(&core.Header{}, nil)
+		mockReader.EXPECT().HeadsHeader().Return(&core.Header{}, nil)
+		mockReader.EXPECT().HighestBlockHeader().Return(nil)
+
+		syncing, err := handler.Syncing()
+		assert.Nil(t, err)
+		assert.Equal(t, &rpc.SyncState{False: &defaultState}, syncing)
+	})
+	t.Run("block height is greater than highest block", func(t *testing.T) {
+		mockReader.EXPECT().StartingBlockNumber().Return(&startingBlock)
+		mockReader.EXPECT().BlockHeaderByNumber(startingBlock).Return(&core.Header{}, nil)
+		mockReader.EXPECT().HeadsHeader().Return(&core.Header{Number: 1}, nil)
+		mockReader.EXPECT().HighestBlockHeader().Return(&core.Header{Number: 0})
+
+		syncing, err := handler.Syncing()
+		assert.Nil(t, err)
+		assert.Equal(t, &rpc.SyncState{False: &defaultState}, syncing)
+	})
+	t.Run("syncing", func(t *testing.T) {
+		mockReader.EXPECT().StartingBlockNumber().Return(&startingBlock)
+		mockReader.EXPECT().BlockHeaderByNumber(startingBlock).Return(&core.Header{Hash: &felt.Zero}, nil)
+		mockReader.EXPECT().HeadsHeader().Return(&core.Header{Number: 1, Hash: new(felt.Felt).SetUint64(1)}, nil)
+		mockReader.EXPECT().HighestBlockHeader().Return(&core.Header{Number: 2, Hash: new(felt.Felt).SetUint64(2)})
+
+		expectedSyncing := &rpc.SyncState{
+			Status: &rpc.SyncStatus{
+				StartingBlockHash:   &felt.Zero,
+				StartingBlockNumber: "0x" + strconv.FormatUint(startingBlock, 16),
```

Comment:
I think this is a common enough pattern that we can define a `HexUint64` type and implement a custom MarshalJSON function for it.

Reply:
I probably don't fully understand your comment. But not really sure we need a custom `HexUint64` type. I have created a `Uint64ToHexStr` function [here](https://github.com/NethermindEth/juno/pull/708/files#diff-df2395a5aa8fcd186ed6c403ff5b4c573d69350914e6bca767cd9b559cd41afeR5)

Reply:
Now that I think about it, printing block numbers in base 10 makes more sense to me, WDYT?

Reply:
my bad then, thanks!